sway/main.c

```diff
@@ -319,6 +319,8 @@ int main(int argc, char **argv) {
 	if (!getenv("XDG_RUNTIME_DIR") && optind == argc) {
 		fprintf(stderr,
 				"XDG_RUNTIME_DIR is not set in the environment. Aborting.\n");
+		fprintf(stderr,
+				"Please check that user is member of seat group and that the seatd.service is enabled.\n");
```

Comment:
`seatd` **does not** create or set `XDG_RUNTIME_DIR`.

I was faced with the error in the line above when setting up my current system, and pushing me to check `seatd` would have been a distraction from the real issue.

Reply:
I think providing a little solution would be best, just adding this to the ~/.profile does the trick: `[ -d "$XDG_RUNTIME_DIR" ] || { mkdir -p "$XDG_RUNTIME_DIR" && chmod 0700 "$XDG_RUNTIME_DIR"; }`

Reply:
@xplshn This solution assumes that the environment variable is set; if the directory does not exist, it is very likely that the variable is unset too.

Reply:
Yes, just gotta set the variable. Can be done in the same line. You can even use `mktemp` or do it like I do in my ~/.profile (since I use Alpine)
```
export XDG_RUNTIME_DIR="${XDG_RUNTIME_DIR:-/tmp/.$(id -u)-runtime-dir}"
[ -d "$XDG_RUNTIME_DIR" ] || { mkdir -p "$XDG_RUNTIME_DIR" && chmod 0700 "$XDG_RUNTIME_DIR"; }

```

![image](https://github.com/user-attachments/assets/dad06e5f-dcfe-40c1-8a65-b69df5927002)